pennylane/_device.py

```diff
@@ -36,6 +36,48 @@
 from pennylane.wires import Wires, WireError
 
 
+def _process_shot_sequence(shot_list):
+    """Process the shot sequence, to determine the total
+    number of shots and the shot vector.
+
+    Args:
+        shot_list (Sequence[int, tuple[int]]): sequence of non-negative shot integers
+
+    Returns:
+        tuple[int, list[int, tuple[int]]]: A tuple containing the total number
+        of shots, as well as the shot vector.
+
+    **Example**
+
+    >>> shot_list = [3, 1, 2, 2, 2, 2, 6, 1, 1, 5, 12, 10, 10]
+    >>> _process_shot_sequence(shot_list)
+    (57, [3, 1, (2, 4), 6, (1, 2), 5, 12, (10, 2)])
+
+    The total number of shots (57), and a sparse representation of the shot
+    sequence is returned, where tuples indicate the number of times a shot
+    integer is repeated.
+    """
+    if all(isinstance(s, int) for s in shot_list):
+
+        if len(set(shot_list)) == 1:
+            # All shots are identical; represent the shot vector
+            # in a sparse format.
+            shot_vector = [(shot_list[0], len(shot_list))]
+            total_shots = shot_list[0] * len(shot_list)
+        else:
+            # Iterate through the shots, and group consecutive identical shots
+            split_at_repeated = np.split(shot_list, np.where(np.diff(shot_list) != 0)[0] + 1)
```

Comment:
```suggestion
            split_at_repeated = np.split(shot_list, np.diff(shot_list).nonzero()[0] + 1)
```
As per the Note here:
https://numpy.org/doc/stable/reference/generated/numpy.where.html?highlight=where#numpy.where